Library/Homebrew/update_migrator.rb

```diff
@@ -0,0 +1,39 @@
+require "update_migrator/cache_entries_to_double_dashes"
+require "update_migrator/cache_entries_to_symlinks"
+require "update_migrator/legacy_cache"
+require "update_migrator/legacy_keg_symlinks"
+require "update_migrator/legacy_repository"
```

Comment:
I think it would be nicer to have all the `UpdateMigrator` logic in a single file.

Reply:
Why? That would solve the problem of having a big file by moving it to another big file.

Reply:
That's how we define modules in the Homebrew codebase. I think having two bigger files is still an improvement over one. Also, I think the migration logic is sensible to keep separate from the update logic itself

Reply:
> I think having two bigger files is still an improvement over one.

And having separate, easily readable, files for each migration is even more of an improvement.

> Also,  I think the migration logic is sensible to keep separate from the update logic itself.

What do you mean by that? That's what this is doing.

Reply:
> > Also, I think the migration logic is sensible to keep separate from the update logic itself.
>
> What do you mean by that? That's what this is doing.

I mean moving to a separate file is a sufficient improvement to still be worth doing.

> And having separate, easily readable, files for each migration is even more of an improvement.

When it's different to the majority of the Homebrew codebase: it makes things harder to follow rather than easier. For example, if I want to find the definition of `UpdateMigrator` I need to look at multiple files rather than one and there's no longer a mapping between the test filename and the implementation filename.